examples/Example.h

```diff
@@ -0,0 +1,537 @@
+/*
+ * Copyright (c) 2024 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "config.h"
+
+#include <vector>
+#include <fstream>
+#include <iostream>
+#include <thread>
+#include <thorvg.h>
+#include <SDL.h>
+#include <SDL_syswm.h>
+#ifdef _WIN32
+    #define WIN32_LEAN_AND_MEAN
+    #include <windows.h>
+    #ifndef PATH_MAX
+        #define PATH_MAX MAX_PATH
+    #endif
+#else
+    #include <dirent.h>
+    #include <unistd.h>
+    #include <limits.h>
+    #include <sys/stat.h>
+#endif
+
+#ifdef THORVG_WG_RASTER_SUPPORT
+    #include <webgpu/webgpu.h>
+    #if defined(SDL_VIDEO_DRIVER_COCOA)
+        #include <Cocoa/Cocoa.h>
+        #include <QuartzCore/CAMetalLayer.h>
+    #endif
+#endif
+
+
+using namespace std;
+
+/************************************************************************/
+/* Common Template Code                                                 */
+/************************************************************************/
+
+namespace tvgexam
+{
+
+bool verify(tvg::Result result);
+
+struct Example
+{
+    virtual bool content(tvg::Canvas* canvas, uint32_t w, uint32_t h) = 0;
+    virtual bool update(tvg::Canvas* canvas, uint32_t elapsed) { return false; }
+    virtual void populate(const char* path) {}
+    virtual ~Example() {}
+
+    void scandir(const char* path)
+    {
+        char buf[PATH_MAX];
+
+        //real path
+    #ifdef _WIN32
+        auto rpath = _fullpath(buf, path, PATH_MAX);
+    #else
+        auto rpath = realpath(path, buf);
+    #endif
+
+        //open directory
+    #ifdef _WIN32
+        WIN32_FIND_DATA fd;
+        HANDLE h = FindFirstFileEx((path + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);
```

Comment:
```
../examples/Example.h:85:42: error: invalid operands of types 'const char*' and 'const char [3]' to binary 'operator+'
   85 |         HANDLE h = FindFirstFileEx((path + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);
      |                                     ~~~~ ^ ~~~~~
      |                                     |      |
      |                                     |      const char [3]
      |                                     const char*
```
Should be some thing like?
`HANDLE h = FindFirstFileEx((std::string(path) + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);`